### DISCUSSION TITLE
Why does the root directory have such a name?

### DISCUSSION THREAD

**Discussion by Ingood5** [d4rken-org/sdmaid-se#535] [Q&A] [answered]
Почему в SD Maid SE корневой каталог называется tree/primary/? В проводниках и в SD Maid 1 корневой каталог называется storage/emulated/0. 

> Why is the root directory called tree/primary/ in SD Maid SE? In explorers and in SD Maid 1, the root directory is called storage/emulated/0.

**Comment by d4rken** [comment, marked as answer]
It depends on the access mesthod.

It's the same location, but different access methods.

`/storage/emulated/0` is the path when using Java File API.
`/tree/primary `is the path when using the Storage Access Framework.

I could fake the name to make it a bit easier for the user, but when creating exclusion I would have to use the correct name...

Hm. When using exclusions I would need to check both paths, an exclusion might not work if the access methods change :thinking: 

PS: Is it possible that you use deepl.com to have English titles for tickets?
